src/diagnostics.rs

```diff
@@ -58,7 +59,7 @@ impl MachineStopType for TerminationInfo {}
 /// Miri specific diagnostics
 pub enum NonHaltingDiagnostic {
     CreatedPointerTag(NonZeroU64),
-    PoppedPointerTag(Item),
+    PoppedPointerTag(Item, Option<(SbTag, AccessKind)>),
```

Comment:
Could you add a doc comment here similar to what you have for `check_protector`?